source/lambda/documentprocessor/lambda_function.py

```diff
@@ -127,6 +127,12 @@ def processRecord(record, syncQueueUrl, asyncQueueUrl, errorHandlerQueueUrl):
         request['errorHandlerQueueUrl'] = errorHandlerQueueUrl
         processRequest(request)
 
+        # send for barcode processing to the same queue, not matter if its pdf or png
+        if syncBarcodeQueueUrl:
+            request['syncQueueUrl'] = syncBarcodeQueueUrl
```

Comment:
separate function: I refactored the process request function to avoid duplicate code, filetype distinguishing is happening before sending to the queues.

Reply:
I added support for multi pipeline processing. The state managements is done over a dynamodb column list (documentPipesFinished). The documen gets marked as SUCCEEDED only if all pipelines get finished.